### ISSUE TITLE
Add PayPal button on checkout

### ISSUE THREAD

**Issue by elpollo12** [flaviocopes/grav-plugin-shoppingcart-paypal#2]
I'm trying to add PayPal as payment method in the checkout but can not find how to do it from the form. When I click on the "Go and Pay " button performs no action and returns to the checkout . I've already enabled paypal plugin in the Control Panel. 

Shopping Cart v1.1.0-beta.8

Shopping Cart PayPal Express Addon v1.1.0-beta.2 test release

I appreciate the help. Thanks.

**Comment by flaviocopes** [comment]
What's the content of your `user/config/plugins/shoppingcart.yaml`?

Since you are using the beta, it probably has a problem with the form definition, as due to a problem in Grav's own config I had to revert providing a full checkout form definition. 

The plan is to provide a couple samples to choose from.

Try filling the checkout form in this way, in `user/config/plugins/shoppingcart.yaml`

```yaml
checkout_form:
  name: checkout
  fields:
    -
        type: display
        content: PLUGIN_SHOPPINGCART.CHECKOUT_HEADLINE_YOUR_PERSONAL_DETAILS
        markdown: true

    -
        name: firstname
        type: text
        label: PLUGIN_SHOPPINGCART.CHECKOUT_FIRST_NAME
        classes: js__billing__firstname
        validate:
            required: true
    -
        name: lastname
        type: text
        label: PLUGIN_SHOPPINGCART.CHECKOUT_LAST_NAME
        classes: js__billing__lastname
        validate:
            required: true

    -
        name: email
        type: email
        label: PLUGIN_SHOPPINGCART.CHECKOUT_EMAIL
        classes: js__billing__email
        validate:
            required: true

    -
        type: display
        content: PLUGIN_SHOPPINGCART.CHECKOUT_HEADLINE_YOUR_ADDRESS
        markdown: true

    -
        name: address
        type: text
        label: PLUGIN_SHOPPINGCART.CHECKOUT_ADDRESS
        classes: js__billing__address
        validate:
            required: true

    -
        name: telephone
        type: text
        label: PLUGIN_SHOPPINGCART.CHECKOUT_PHONE
        classes: js__billing__telephone
        validate:
            required: true

    -
        name: city
        type: text
        label: PLUGIN_SHOPPINGCART.CHECKOUT_CITY
        classes: js__billing__city
        validate:
            required: true

    -
        name: zip
        type: text
        label: PLUGIN_SHOPPINGCART.CHECKOUT_ZIP
        classes: js__billing__zip
        validate:
            required: true

    -
        name: country
        type: select
        label: PLUGIN_SHOPPINGCART.CHECKOUT_COUNTRY
        classes: js__billing__country
        id: js__billing__country
        validate:
            required: true

    -
        name: state
        type: select
        label: PLUGIN_SHOPPINGCART.CHECKOUT_STATE
        classes: js__billing__state
        outerclasses: js__billing__state__control
        id: js__billing__state

    -
        name: province
        type: text
        label: PLUGIN_SHOPPINGCART.CHECKOUT_PROVINCE
        classes: js__billing__province
        outerclasses: js__billing__province__control

    -
        type: display
        content: PLUGIN_SHOPPINGCART.CHECKOUT_SHIPPING_METHOD
        ignore: true
        markdown: true

    -
        name: shipping_method
        type: select
        label: PLUGIN_SHOPPINGCART.CHECKOUT_CHOOSE_SHIPPING_METHOD_DESC
        classes: js__shipping__method
        outerclasses: js__checkout-choose-shipping-block
        id: js__shipping__method
        validate:
            required: true

    -
        type: display
        content: PLUGIN_SHOPPINGCART.CHECKOUT_PAYMENT_METHOD
        markdown: true
        ignore: true
        outerclasses: js__checkout-choose-payment-block-title

    -
        name: payment_method
        type: select
        ignore: true
        label: PLUGIN_SHOPPINGCART.CHECKOUT_CHOOSE_PAYMENT_METHOD_DESC
        classes: js__payment__method
        outerclasses: js__checkout-choose-payment-block
        id: js__payment__method

    -
        name: agree_to_terms
        type: checkbox
        ignore: true
        label: PLUGIN_SHOPPINGCART.TERMS_AND_CONDITIONS
        validate:
            required: true

  buttons:
    -
      type: submit
      value: PLUGIN_SHOPPINGCART.CHECKOUT_BUTTON_PAY
```

**Comment by elpollo12** [comment]
Thanks for your help Flavio. This is the code of my shoppingcart.yaml

```
enabled: true
general:
  currency: USD
  default_country: US
  product_taxes: excluded
  load_js_globally: true
shipping:
  methods:
    -
      name: 'All countries'
      allowed_countries:
        - '*'
      price: '0'
urls:
  checkout_url: /checkout
  save_order_url: /save_order
  order_url: /order
ui:
  currency_symbol_position: before
  remove_cents_if_zero: true
  use_own_css: true
  image_size_product: 400
  image_size_cart: 50
cart:
  allow_editing_quantity_from_cart: true
  timeout: 180
  add_shipping_and_taxes_cost_to_total: true
  maximum_total_quantity_value:
  add_product_thumbnail: true
countries:
  -
    name: US
    allow: '1'
    tax_percentage: '10'
  -
    name: '*'
    allow: '1'

checkout_form:
  name: checkout
  fields:
    -
        type: display
        content: PLUGIN_SHOPPINGCART.CHECKOUT_HEADLINE_YOUR_PERSONAL_DETAILS
        markdown: true

    -
        name: firstname
        type: text
        label: PLUGIN_SHOPPINGCART.CHECKOUT_FIRST_NAME
        classes: js__billing__firstname
        validate:
            required: true
    -
        name: lastname
        type: text
        label: PLUGIN_SHOPPINGCART.CHECKOUT_LAST_NAME
        classes: js__billing__lastname
        validate:
            required: true

    -
        name: email
        type: email
        label: PLUGIN_SHOPPINGCART.CHECKOUT_EMAIL
        classes: js__billing__email
        validate:
            required: true

    -
        type: display
        content: PLUGIN_SHOPPINGCART.CHECKOUT_HEADLINE_YOUR_ADDRESS
        markdown: true

    -
        name: address
        type: text
        label: PLUGIN_SHOPPINGCART.CHECKOUT_ADDRESS
        classes: js__billing__address
        validate:
            required: true

    -
        name: telephone
        type: text
        label: PLUGIN_SHOPPINGCART.CHECKOUT_PHONE
        classes: js__billing__telephone
        validate:
            required: true

    -
        name: city
        type: text
        label: PLUGIN_SHOPPINGCART.CHECKOUT_CITY
        classes: js__billing__city
        validate:
            required: true

    -
        name: zip
        type: text
        label: PLUGIN_SHOPPINGCART.CHECKOUT_ZIP
        classes: js__billing__zip
        validate:
            required: true

    -
        name: country
        type: select
        label: PLUGIN_SHOPPINGCART.CHECKOUT_COUNTRY
        classes: js__billing__country
        id: js__billing__country
        validate:
            required: true

    -
        name: state
        type: select
        label: PLUGIN_SHOPPINGCART.CHECKOUT_STATE
        classes: js__billing__state
        outerclasses: js__billing__state__control
        id: js__billing__state

    -
        name: province
        type: text
        label: PLUGIN_SHOPPINGCART.CHECKOUT_PROVINCE
        classes: js__billing__province
        outerclasses: js__billing__province__control

    -
        type: display
        content: PLUGIN_SHOPPINGCART.CHECKOUT_SHIPPING_METHOD
        ignore: true
        markdown: true

    -
        name: shipping_method
        type: select
        label: PLUGIN_SHOPPINGCART.CHECKOUT_CHOOSE_SHIPPING_METHOD_DESC
        classes: js__shipping__method
        outerclasses: js__checkout-choose-shipping-block
        id: js__shipping__method
        validate:
            required: true

    -
        type: display
        content: PLUGIN_SHOPPINGCART.CHECKOUT_PAYMENT_METHOD
        markdown: true
        ignore: true
        outerclasses: js__checkout-choose-payment-block-title

    -
        name: payment_method
        type: select
        ignore: true
        label: PLUGIN_SHOPPINGCART.CHECKOUT_CHOOSE_PAYMENT_METHOD_DESC
        classes: js__payment__method
        outerclasses: js__checkout-choose-payment-block
        id: js__payment__method

    -
        name: agree_to_terms
        type: checkbox
        ignore: true
        label: PLUGIN_SHOPPINGCART.TERMS_AND_CONDITIONS
        validate:
            required: true

  buttons:
    -
      type: submit
      value: PLUGIN_SHOPPINGCART.CHECKOUT_BUTTON_PAY
```

The method of payment field is still empty. Also I upgrade the shopping cart (v1.1.0-beta.10) and paypal (1.0.2). When I click on the "Go and Pay " button performs no action and returns to the checkout.

**Comment by flaviocopes** [comment]
Should now be fixed in Grav 1.1.x
